config/routes.rb

```diff
@@ -1,3 +1,13 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  # define api endpoints
+  namespace :api, dafaults: {format: :json}, 
+                  contraints: {subdomain: 'api'},
+                  paths: '/' do
```

Comment:
Tells Rails that there is a folder under app/controllers named 'api', it also defines the path to start from root '/' and then tells rails that the default format coming from these routes will be json

Reply:
contraints: {subdomain: 'api'}

defines the possible routes rails will expect,
- api.example.com/
instead of
- example.com/api/